Updating Hydra to Version to 0.7.0

emarinier opened the pull request.

apetkau reviewed (changes requested):
Thanks so much for submitting this Eric. A few comments:

1. The tests fail for me with the following message:

    ```
    "Output output_aa_mt:  different than expected\nExpected text matching expression '#CHROM\\sGENE\\sTYPE\\sWILDTYPE\\sPOS\\sMUTANT\\sFILTER\\sMUTANT_FREQ\\sCOVERAGE\\sINFO' in output ('')"
    ```
    
    I'm wondering if the source of the issue is here:
    
    https://github.com/phac-nml/galaxy_tools/blob/84c50ccdfc230aef0b326f06fd08cf359b396916/tools/quasitools/hydra.xml#L184
    
    Since it looks like it's expected a file to end in `.hmcf` but when I ran hydra on the command-line I see a file named `.aavf`.

2. I'm wondering if it's possible to move the version number and other common elements (like `<citations>` from each individual XML file to a `macros.xml` file which can then be imported by each individual Galaxy tool.

    As an example, consider the `mlst` tool in Galaxy. There exists a file in this repo `macros.xml` that contains https://github.com/galaxyproject/tools-iuc/blob/master/tools/mlst/macros.xml

    In other words, it contains a `<token name="@VERSION@">2.19.0</token>` entry that defines the underlying dependency version in conda as well as the common `<requirements>` and `<citations>` sections.
    
    These are all imported into each tool XML file (`mlst.xml`) like the following: https://github.com/galaxyproject/tools-iuc/blob/master/tools/mlst/mlst.xml#L6-L7

    This means you can use the token `@VERSION@` in the `mlst.xml` file like https://github.com/galaxyproject/tools-iuc/blob/master/tools/mlst/mlst.xml#L1
    
    And you can include the common `<citations>` section like https://github.com/galaxyproject/tools-iuc/blob/master/tools/mlst/mlst.xml#L359

    Since there's a lot of different tools in this repository it could help make sure they are all in-sync with each other.
    
    Here's some additional information on macros in Galaxy tools: https://planemo.readthedocs.io/en/latest/writing_advanced.html#imported-macros

emarinier commented:
Thanks Aaron. I'm guessing this related to why HyDRA in particular wasn't updated to 0.7.0, when all the other tools were. There was a change in output format for HyDRA (to aavf), so it makes sense that this error is coming up.

And yeah, I can work on converting them to use the macro format.

emarinier commented:
- Converted the hmcf format to aavf and updated the related tests and descriptions in hydra.xml
- Added the macro.xml format to all files
- Changed the symbolic linking being done for several tools, which was causing tool errors.

emarinier commented on tools/quasitools/aacoverage.xml:
```diff
@@ -1,13 +1,15 @@
 <tool id="aacoverage" name="Amino Acid Coverage" version="0.7.0">
```
Wooops!

apetkau replied:
No problem. This looks great. Just testing it out now :)

apetkau commented on tools/quasitools/aacoverage.xml:
```diff
@@ -1,13 +1,15 @@
+        ln -s $input_bam.metadata.bam_index '${input_bam.name}.bai' &&
 
-        quasitools aacoverage $input_bam $ref_file $input_genes -o output.csv
+        quasitools aacoverage ${input_bam.name} $ref_file $input_genes -o output.csv
```
It's also recommended to add quotes around file names so that e.g., spaces in file names on the command-line won't mess with things. So this could be changed to `'${input_bam.name}'`

emarinier replied:
Ah, good catch. I guess the `.name` part can introduce the spaces, whereas the other parameters will be all `.dat`.

emarinier commented on tools/quasitools/aacoverage.xml:
```diff
@@ -1,13 +1,15 @@
-<tool id="aacoverage" name="Amino Acid Coverage" version="0.7.0">
+<tool id="aacoverage" name="Amino Acid Coverage" version="@TOOL_VERSION@">
```
Thanks Aaron. That makes sense. I think I remember doing something like this in the past.

I've updated the tools; hopefully they're good now!

apetkau replied:
Yep, it all looks great. Thanks so much.
